### PR TITLE
Add tests for command-line parser

### PR DESCRIPTION
**Pull request by olafurpg**
Previously, we didn't have any tests for how we parsed command-line
flags. This was problematic because the tests didn't catch breaking
changes in seemingly innocent pull requests like https://github.com/sourcegraph/lsif-typescript/pull/100
that rename a variable. Now, we have tests for the command-line parser
so that we hopefully catch more unintentional regressions in the future.

### Test plan

See new `CommandLineOptions.test.ts` tests.

### CODE REVIEW

File: src/inferTsconfig.ts [hunk 1]
```diff
@@ -19,7 +19,7 @@ export const noJsConfig = '{}'
  * If the directory contains at least one `*.{ts,tsx}` file then the config will be empty (`{}`).
  * If the directory doesn't contains one `*.{ts,tsx}` file then the config will
  */
-export function inferTsConfig(projectPath: string): string {
+export function inferTsconfig(projectPath: string): string {
```
Comment (olafurpg) [inline src/inferTsconfig.ts:22]: I apologize for mixing this unrelated diff into the PR. It's quite painful to correct the capitalization of files on macOS and I didn't feel like doing it again to have a cleaner PR

File: src/CommandLineOptions.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+/** Configuration options to index a single TypeScript project. */
+export interface ProjectOptions extends MultiProjectOptions {
+  projectRoot: string
+  projectDisplayName: string
+  writeIndex: (index: lsif.lib.codeintel.lsiftyped.Index) => void
+}
+
+export function commanderCommand(
+  indexAction: (projects: string[], otpions: MultiProjectOptions) => void
+): Command {
```
Comment (varungandhi-src) [inline src/CommandLineOptions.ts]:
```suggestion
export function mainCommand(
  indexAction: (projects: string[], options: MultiProjectOptions) => void
): Command {
```

Maybe name the function to be something more generic/describing its use, rather than the implementation? The fact that we're using the `commander` library is somewhat incidental.

File: src/CommandLineOptions.test.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
+import { test } from 'uvu'
+import * as assert from 'uvu/assert'
+
+import { commanderCommand, MultiProjectOptions } from './CommandLineOptions'
+
+function checkIndexParser(
+  args: string[],
+  expectedOptions: Partial<MultiProjectOptions>,
```
Comment (varungandhi-src) [inline src/CommandLineOptions.test.ts:8]: Oh wow, I didn't know this was a thing you could do so easily in TypeScript. (The `Partial<>` type)

File: src/CommandLineOptions.ts [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    .option('--cwd', 'the working directory', process.cwd())
+    .option('--yarn-workspaces', 'whether to index all yarn workspaces', false)
+    .option(
+      '--infer-tsconfig',
+      "whether to infer the tsconfig.json file, if it's missing",
+      false
+    )
+    .option('--output', 'path to the output file', 'dump.lsif-typed')
+    .option('--no-progress-bar', 'whether to disable the progress bar')
+    .argument('[projects...]')
```
Comment (varungandhi-src) [inline src/CommandLineOptions.ts]: Now I'm wondering... is there a difference between:

```
lsif-typescript index --cwd /some/path
vs
lsif-typescript index /some/path
```

?

File: src/CommandLineOptions.ts [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+}
+
+export function commanderCommand(
+  indexAction: (projects: string[], otpions: MultiProjectOptions) => void
+): Command {
+  const command = new Command()
+  command
+    .name('lsif-typescript')
+    .version(packageJson.version)
+    .description('LSIF indexer for TypeScript and JavaScript')
```
Comment (varungandhi-src) [inline src/CommandLineOptions.ts]: Maybe add a line like:

```
For usage examples, see https://github.com/sourcegraph/lsif-typescript/blob/main/README.md
```

**Review by varungandhi-src — APPROVED** [review]

File: src/CommandLineOptions.ts (hunk 2 above)
Comment (olafurpg) [reply src/CommandLineOptions.ts]: Great point, it was a bad name, renamed to `mainCommand`.

File: src/CommandLineOptions.ts (hunk 4 above)
Comment (olafurpg) [reply src/CommandLineOptions.ts]: It's good you asked because it surfaced a bug in how we parse the `--cwd` flag. I fixed the bug and added a cli test for `--cwd`. There is a difference between those two:

- `index --cwd /path` generates `/path/dump.lsif-typed`
- `index /path` generates `$PWD/dump.lsif-typed`

File: src/CommandLineOptions.ts (hunk 5 above)
Comment (olafurpg) [reply src/CommandLineOptions.ts]: Good idea, done.

File: src/CommandLineOptions.test.ts (hunk 3 above)
Comment (olafurpg) [reply src/CommandLineOptions.test.ts:8]: It's very handy for cases like this! TypeScript includes several utility types https://www.typescriptlang.org/docs/handbook/utility-types.html#partialtype but you can go a bit overboard with them so use with care.
